lib/ui/events.js

```diff
@@ -0,0 +1,317 @@
+        border : { fg: 'white' },
+        header : { fg: 'grey' },
+        cell   : { fg: 'white', selected: { bg: 'blue' } },
+      }
+    });
+
+
+    // do not render when detached. 
+    events_table._orender = events_table.render;
+    events_table.render = function () {
+      if (this.parent == null) {
+        return;
+      }  
+      this._orender();
+    };
+
+    // work around to preserve selection and scrolling
+    events_table.on('detach', () => {
+      const { selected, childBase, childOffset } = events_table;
+      events_table.once('attach', () => {
+        events_table.select(selected);
+        Object.assign(events_table, { childBase, childOffset });
+        events_table.scrollTo(selected);
+      });
+    });
+
+    events_table.on('key C-z', () => {
+      cancellations.run(key);
+      navbar.remove(key);
+    });
+
+    const { promise, cancellation } = client.events(namespace).get({ cancellable : true });
+    cancellations.add(key, cancellation);
+    until(promise)
+    .spin(s => events_table.setLabel(`${s} ${key}`))
+    .succeed(_ => events_table.setLabel(`${key}`))
+    .then(response => {
+      const events_list = JSON.parse(response.body.toString('utf8'));
+      events_table.resource_version = events_list.metadata.resourceVersion;
+      this.fill_events_table(events_list, events_table, cancellations, key);
+    })
+    .then(_ =>{
+      const { promise, cancellation } = client.event(namespace)
+      .watch(events_table.resource_version)
+      .get({ generator: update_events_table.bind({ events_table, debug, highlightEvent })})
+      cancellations.add(key, cancellation);
+      return promise;
+    }).catch(error => {
+      debug.log(`{red-fg}Error creating Events tab: ${error}{/red-fg}`);
+      if (error.response.statusCode == '403') {
+        const unauthorized_msg = `{red-fg}It seems you don't have the rights to fetch ${key}. Please contact your Cluster administrator.{/red-fg}`;
+        debug.log(unauthorized_msg);
+        events_table.setData([[unauthorized_msg]]);
+      }
+    });
+
+    events_table.on('select', (item, i) => {
+      if (i === 0 || events_table.uid === events_table.rows[i].uid) return;
+      this.describe_event(i);
+    });
+
+    this.describe_event = function(i) {
+      event_description.setContent("");
+      const namespace = events_table.rows[i].namespace;
+      const name = events_table.rows[i].name;
+      events_table.uid = events_table.rows[i].uid;
+      const { promise, cancellation } = client.describe_event(namespace, name).get({ cancellable : true });
+      cancellations.add(key, cancellation);
+      until(promise)
+      .spin(s => event_description.setLabel(`${s} Event`))
+      .succeed(_ => event_description.setLabel(`Event ${events_table.rows[i][2]} {inverse}${events_table.rows[i][4]}{/inverse}`))
+      .then(response => {
+        const events_description = yaml.stringify(JSON.parse(response.body.toString('utf8')));
```

Comment:
Can't the event be requested as Yaml so that we can avoid having to depend on that extra dependency?

Reply:
I was expecting the same, however when debugging the `oc` and `kubectl` clients using --v=99 I see that the server always returns JSON format even with `-o yaml` flag so I don't think it's possible

Reply:
I'm not sure `kubectl --v=...` is reliable, as it is the same for other resources like pods. It seems it output result from curl with `Accept: application/json`, so I wonder if the server actually serves Yaml when provided with the correct header.

Reply:
Yes it does! Dependency removed!